### PR TITLE
Added backoff retry mechanism.

### PR DESCRIPTION
**Pull request by keithballdotnet**
I needed this.  Maybe someone else does too.

### CODE REVIEW

**Comment by keithballdotnet** [comment]
Do I need to squash the commits or are you happy with that wasted commit?

File: retry.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+		sleepDuration := time.Duration(sleepTime) * time.Millisecond
+		time.Sleep(sleepDuration)
+	}
+
+	return err
+}
+
+// Option ...
+type Option func(*Options)
```
Comment (jeevatkm) [inline retry.go]: Please move type definition after the import and implementation comes after that. To keep consistent across the library code.

File: retry.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// Options ...
+type Options struct {
```
Comment (jeevatkm) [inline retry.go]: Please move type definition after the import and implementation comes after that. To keep consistent across the library code.

File: retry_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package resty
+
+import (
+	"errors"
+	"testing"
+)
+
+func Test_Backoff_Success(t *testing.T) {
```
Comment (jeevatkm) [inline retry_test.go]: Please remove the underscore.

File: retry_test.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+			return errors.New("Not yet got the number we're after...")
+		}
+		return nil
+	})
+
+	assertError(t, retryErr)
+	assertEqual(t, externalCounter, attempts)
+}
+
+func Test_BackoffTenAttempts_Success(t *testing.T) {
```
Comment (jeevatkm) [inline retry_test.go]: Please remove the underscore.

**Pull request closed without merging**